lib/ibm/cloud/sdk_http/sdk_response.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require 'json'
+
+module IBM
+  module Cloud
+    # Module holds basic HTTP functionality.
+    module SDKHTTP
+      # Encapsulate the HTTP response.
+      # @param response [HTTP::Response] The HTTP response object.
+      class SDKResponse
+        def initialize(response)
+          @response = response
```

Comment:
A lot of the methods below use the safe nav operator on response, but is it possible to have a nil response?  If not, should be just raise ArgumentError here if the response is nil and then avoid the safe navigation operator below?

Reply:
I'm doing this because I can't be sure of the object that is being queried. It comes from an external library. It is easier to test for nil than to catch exceptions in client code. If I can get around to raising SDK specific exceptions then maybe an exception is better to stop the client code's execution. Otherwise we may end up with some incomprehensible message for anyone not familiar with the inner workings of the HTTP library.